lib/home/widgets/route_card.dart

```diff
@@ -59,28 +67,41 @@ class _RouteCardState extends State<RouteCard> {
                     Padding(
                       padding: const EdgeInsets.only(bottom: 8),
                       child: Text(
-                        "${widget.route.type.label}, ${widget.route.grade}",
+                        "${ClimbTypeEnum.values[widget.route.climbTypeId].label}, ${widget.route.grade}",
                         style: const TextStyle(
                           fontSize: 14,
                         ),
                       ),
                     ),
                     Padding(
                       padding: const EdgeInsets.only(bottom: 8),
-                      child: Wrap(
-                        runSpacing: -10,
-                        spacing: 4,
-                        children: widget.route.tags
-                            .map((element) => Chip(
-                                  label: Text(element.label),
-                                ))
-                            .toList(),
-                      ),
+                      child: FutureBuilder<List<int>>(
+                          future: db.routeTagDao
+                              .getTagIdsByRouteId(widget.route.id),
+                          builder: (BuildContext context,
+                              AsyncSnapshot<List<int>> snapshot) {
+                            if (snapshot.hasData) {
```

Comment:
I feel like this could be changed or simplified. Perhaps move the conditional to the children.
ex:
`chjldren: snapshot.hasData ? map(...) : []`

I think this would simplify things and reduce code reuse. There may also be a cleaner way to do this with null safety, perhaps using the `??` operator to default to `[]` if data is not loaded. What is `snapshot.data` when `snapshot.hasData` is false? Is it null or does the property not even exist?